### ISSUE TITLE
Background prefetch

### ISSUE THREAD

**Issue by Atemu** [TeamNewPipe/NewPipe#4932]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

Loading a videos' page and starting its playback takes a while because various pieces of information have to be fetched from Youtube, often in unconventional, inefficient manners.  
Under good network conditions, it's fast enough but doesn't feel particularly nice. On spotty connections however (which are to be expected on phones), it can often take multiple seconds or even time out under especially bad conditions.

Optimising fetch speed may be possible but another way of improving the UX would be to preemptively fetch information the user might need in the future while they are doing something else. 
At the point the user needs the information and waiting on it becomes the critical path (e.g. clicked on a new video), it would be available instantly.

Considerations:

* Would going off and prefetching info the user didn't explicitly request expose any more information about the user to Google than we already do?
* What data should be prefetched? Just metadata? Stream URLs? The first few segments of a stream?
* How much more memory would this use?
* How long should prefetched data be kept around/when should it be evicted?
* Would Google be fine with like us fetching so many videos or would we look even more suspicious than we already are?
* Which videos' data should be prefetched and whose shouldn't? (e.g. how far down a feed, how many suggestions)
* Receiving and processing additional information that the user may not even need will need more power than only fetching what is necessary. Does it impact battery life significantly or could it be enabled by default?
* Can background prefetching affect the foreground task? (e.g. stuttering while watching, slower initial video load times because other information is prefetched at the same time)

#### Is your feature request related to a problem? Please describe it

Opening a video and starting playback can often take upwards of 5s, even under good network conditions.

#### How will you/everyone benefit from this feature?

Less time wasted waiting on things to load, possibly next to none.

=> Better UX

**Comment by Stypox** [comment]
I don't think this is a good idea. There are already shortcuts in the long press menu to achieve simple actions without having to wait for the buttons to appear, and all of your considerations about slowdowns, issues with Google, higher data and battery use, etc. are big cons.

**Comment by Atemu** [comment]
> There are already shortcuts in the long press menu to achieve simple actions without having to wait for the buttons to appear

Those only cover a few use-cases which don't include the primary purpose of the app: Browsing and watching videos.

> and all of your considerations about slowdowns, issues with Google, higher data and battery use, etc. are big cons.

\*May be. These are all just potential cons which have to be discussed and/or researched for now.

For example, pre-fetching might not weigh in much in terms of power because the power used by (at worst) an extra few dozen seconds of fetching kilobytes information from the internet should be absolutely dwarfed by the power needed for the minutes or hours of streamed video playback that follow it.
